packages/snaps-simulation/src/simulation.ts

```diff
@@ -260,24 +366,42 @@ export async function installSnap<
  * Get the hooks for the simulation.
  *
  * @param options - The simulation options.
- * @param snapFiles - The Snap files.
- * @param snapId - The Snap ID.
- * @param controllerMessenger - The controller messenger.
  * @returns The hooks for the simulation.
  */
-export function getHooks(
+export function getRestrictedHooks(
   options: SimulationOptions,
-  snapFiles: FetchedSnapFiles,
-  snapId: SnapId,
-  controllerMessenger: RootControllerMessenger,
-): MiddlewareHooks {
+): RestrictedMiddlewareHooks {
   return {
     getMnemonic: async () =>
```

Comment:
Where are the restricted hooks for state management defined?

Reply:
https://github.com/MetaMask/snaps/blob/aa5f7a141fcf2f75ffaca0cdefb9a87252ba2c29/packages/snaps-simulation/src/methods/specifications.ts#L90-L104